lib/cli.ex

```diff
@@ -1,12 +1,30 @@
 defmodule Onigumo.CLI do
   @components %{
-    :downloader => Onigumo.Downloader
+    :downloader => Onigumo.Downloader,
   }
 
   def main(argv) do
-    {[], [component]} = OptionParser.parse!(argv, strict: [])
-    {:ok, module} = Map.fetch(@components, String.to_atom(component))
-    root_path = File.cwd!()
-    module.main(root_path)
+    try do
+      {[], [component]} = OptionParser.parse!(argv, strict: [])
+      component
+    rescue
+      _ in [OptionParser.ParseError, MatchError] -> usage_message()
+    else
+      component ->
+        {:ok, module} = Map.fetch(@components, String.to_atom(component))
+        root_path = File.cwd!()
+        module.main(root_path)
+    end
+  end
+
+  defp usage_message() do
+    valid_components = Enum.join(Map.keys(@components), ", ")
+    IO.puts("""
+    Usage: ./onigumo [COMPONENT]
```

Comment:
Maybe the `./onigumo` is too assuming in how the script is called.
```suggestion
    Usage: onigumo [COMPONENT]
```